Simplenote/src/main/java/com/automattic/simplenote/models/Account.java

```diff
@@ -20,13 +21,16 @@ private Account(String key, JSONObject properties) {
         super(key, properties);
     }
 
-    public boolean hasConfirmedAccount() {
-        Object status = getProperty(FIELD_EMAIL_VERIFICATION_STATUS);
+    public boolean hasConfirmedAccount(String email) {
+        Object token = getProperty(FIELD_EMAIL_VERIFICATION_TOKEN);
 
-        if (status == null) {
+        if (token == null) {
             return false;
+        } else if (((String) token).split(":", 2).length > 0) {
+            String emailFromToken = ((String) token).split(":", 2)[0];
+            return emailFromToken.equals(email);
```

Comment:
would it be helpful to create a separate parsing function for the verification token and then compare the meaningful output here?

```java
class EmailVerificationToken {
    public String email;
    public Calendar verifiedAt;

    public boolean isVerified() { … }
    public boolean isExpired() { … }
}


EmailVerificationToken token = EmailVerificationToken.fromString(getProperty(FIELD_EMAIL_VERIFICATION_TOKEN));

return token.isVerified() && token.email.equals(email);
```

Reply:
I thought about making a `Verification` class, but I chose not to since the verification is simple right now.  If we end up adding more to verification like parsing with a public key, then I was going to move everything to a `Verification` class.

Your comment has the `isExpired` method.  Are we planning to expire verification?  If so, what's the logic for that method?

Reply:
> since the verification is simple right now

it was actually the non-obvious parsing code that scared me. having an explanatory comment at a minimum would make it clearer, rather than only showing implicit expectations based on `:`

```java
// expect tokens in the form [email]:[epoch time in seconds as number]:[verification hash]
// e.g. "simplenote@example.com:193478124:akjn3v9z8ja3jasdf=="
```

of course at this point having a class where the logic is contained seems like it'd be similar in the amount of work but more semantically separated and defined, and less likely to result in duplicated parsing of the token in slightly-different ways around the app

> Are we planning to expire verification?

It was my impression this will happen at some point, such as "It's been a year since you confirmed your email account, can you do that again?"

Reply:
I added comments to the methods in 8b97c9c.